src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -479,11 +486,16 @@ private void PurgeOldLogDirectories()
             }
         }
 
-        public static ScriptHost Create(ScriptHostConfiguration scriptConfig = null, ScriptSettingsManager settingsManager = null)
+        public static ScriptHost Create(IScriptHostEnvironment environment, ScriptHostConfiguration scriptConfig = null, ScriptSettingsManager settingsManager = null)
```

Comment:
Can this thing be optional so it can be defaulted? This is the public factory method for ScriptHost, and I'm not sure people will know where to get one from.